rollup/core/classes/RollupCalculator.cls

```diff
@@ -1258,6 +1271,177 @@ public without sharing abstract class RollupCalculator {
     }
   }
 
+  private without sharing class GroupByCalculator extends DelimiterCalculator {
+    private final GroupingFormatter formatter;
+    private final RollupComparer sorter;
+    private final RollupCalculator innerCalculator;
+    private final List<String> fieldNames = new List<String>();
+    private GroupByCalculator(
+      Rollup.Op op,
+      SObjectField opFieldOnCalcItem,
+      SObjectField opFieldOnLookupObject,
+      Rollup__mdt metadata,
+      SObjectField lookupKeyField
+    ) {
+      super(op, opFieldOnCalcItem, opFieldOnLookupObject, metadata, lookupKeyField);
+      this.setupFields();
+      this.formatter = this.metadata.IsTableFormatted__c
+        ? (GroupingFormatter) new TableGroupingFormatter(opFieldOnCalcItem)
```

Comment:
Do you have to do the casting to `GroupingFormatter` here? I was thinking it was optional since `TableGroupingFormatter` implements `GroupingFormatter`, but maybe you still need it?